File: veaiops/schema/base/data_source.py
# Copyright 2025 Beijing Volcano Engine Technology Co., Ltd. and/or its affiliates
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from typing import Dict, List, Optional

from pydantic import BaseModel, ConfigDict, Field


class ZabbixTarget(BaseModel):
    """Zabbix target configuration."""

    model_config = ConfigDict(populate_by_name=True)

    itemid: str = Field(..., description="Zabbix item ID")
    hostname: str = Field(..., description="Zabbix hostname")


class ZabbixTriggerTag(BaseModel):
    """Zabbix trigger tag configuration."""

    model_config = ConfigDict(populate_by_name=True)

    tag: str = Field(..., description="Tag key")
    value: str = Field(..., description="Tag value")


class BaseDataSourceConfig(BaseModel):
    """Base Data Source configuration."""

    model_config = ConfigDict(populate_by_name=True)
    name: Optional[str] = Field(None, description="Data source name")
    connect_name: str = Field(..., description="Connect name for authentication")
    metric_name: str = Field(default="", description="Metric name")


class ZabbixDataSourceConfig(BaseDataSourceConfig):
    """Zabbix data source configuration."""

    targets: list[ZabbixTarget] = Field(default_factory=list, description="Zabbix targets")
    history_type: int = Field(default=0, description="History type")


class CloudDataSourceConfig(BaseDataSourceConfig):
    """Cloud Data Source configuration."""

    region: str = Field(default="", description="Region")
    namespace: str = Field(default="", description="Namespace")
    group_by: Optional[List[str]] = Field(default=None, description="Group by dimensions")


class AliyunDataSourceConfig(CloudDataSourceConfig):
    """Aliyun data source configuration."""

    dimensions: Optional[List[Dict[str, str]]] = Field(default=None, description="Dimensions")


class VolcengineDataSourceConfig(CloudDataSourceConfig):
    """Volcengine data source configuration."""

    sub_namespace: str = Field(default="", description="Sub namespace")
    instances: Optional[List[Dict[str, str]]] = Field(default=None, description="Instance list")

File: frontend/packages/constants/src/datasource.ts
// Copyright 2025 Beijing Volcano Engine Technology Co., Ltd. and/or its affiliates
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/**
 * 数据源类型常量
 */
export const DATASOURCE_TYPES = {
  VOLCENGINE: 'Volcengine',
  ALIYUN: 'Aliyun',
  ZABBIX: 'Zabbix',
} as const;

/**
 * 数据源类型配置（包含翻译和样式）
 */
export const DATA_SOURCE_CONFIG = {
  Zabbix: {
    label: 'Zabbix',
    color: 'blue',
    iconColor: '#1890ff',
  },
  Aliyun: {
    label: '阿里云',
    color: 'orange',
    iconColor: '#fa8c16',
  },
  Volcengine: {
    label: '火山引擎',
    color: 'purple',
    iconColor: '#722ed1',
  },
} as const;

/**
 * 数据源类型选项（用于下拉选择等）
 */
export const DATA_SOURCE_OPTIONS = [
  { label: 'Zabbix', value: 'Zabbix' },
  { label: '阿里云', value: 'Aliyun' },
  { label: '火山引擎', value: 'Volcengine' },
] as const;

/**
 * 数据源类型映射（用于快速查找翻译）
 */
export const DATA_SOURCE_LABELS = {
  Zabbix: 'Zabbix',
  Aliyun: '阿里云',
  Volcengine: '火山引擎',
} as const;

/**
 * 获取数据源类型的中文标签
 */
export function getDataSourceLabel(type: string): string {
  return DATA_SOURCE_LABELS[type as keyof typeof DATA_SOURCE_LABELS] || type;
}

/**
 * 获取数据源类型配置
 */
export function getDataSourceConfig(type: string) {
  return (
    DATA_SOURCE_CONFIG[type as keyof typeof DATA_SOURCE_CONFIG] || {
      label: type,
      color: 'default',
      iconColor: '#666666',
    }
  );
}
